Source/Core/Core/NetPlayServer.cpp

```diff
@@ -2067,9 +2068,17 @@ bool NetPlayServer::SyncCodes()
   }
   // Sync Gecko Codes
   {
+    std::vector<Gecko::GeckoCode> s_codes = Gecko::LoadCodes(globalIni, localIni);
+
+#ifdef USE_RETRO_ACHIEVEMENTS
+    {
+      std::lock_guard lg{AchievementManager::GetInstance().GetLock()};
```

Comment:
This is unnecessary. `FilterApprovedIni` already locks this.

Requiring the caller to acquire a certain lock is in general bad design